src/rust/engine/task_executor/src/lib.rs

```diff
@@ -155,17 +147,16 @@ impl Executor {
   /// by it ends up in the pantsd log as we expect. The latter ensures that when a new workunit
   /// is created it has an accurate handle to its parent.
   ///
-  fn future_with_correct_context<Item, Error, F: Future<Item = Item, Error = Error>>(
-    future: F,
-  ) -> impl Future<Item = Item, Error = Error> {
+  fn future_with_correct_context<F: Future>(future: F) -> impl Future<Output = F::Output> {
     let logging_destination = logging::get_destination();
     let workunit_parent_id = workunit_store::get_parent_id();
-    future::lazy(move || {
+    future::lazy(move |_| {
       logging::set_destination(logging_destination);
       if let Some(parent_id) = workunit_parent_id {
         workunit_store::set_parent_id(parent_id);
       }
       future
     })
+    .then(|f| f)
```

Comment:
Why?

Reply:
Because `future::lazy` no longer takes a future: it takes a block that produces a value.

BUT: I think that this could maybe be replaced with an `async` closure...? Will look.

Reply:
Indeed it can. Nice.